self-hosted/src/main/resources/templates/login.html

```diff
@@ -0,0 +1,58 @@
+<!--
+  ~ Copyright 2017 Okta, Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+<head>
+    <title>Login</title>
+    <!--/*/ <th:block th:include="head :: head"/> /*/-->
+
+    <script src="https://global.oktacdn.com/okta-signin-widget/5.5.2/js/okta-sign-in.min.js" type="text/javascript"></script>
+    <link href="https://global.oktacdn.com/okta-signin-widget/5.5.2/css/okta-sign-in.min.css" type="text/css" rel="stylesheet"/>
+
+</head>
+<body class="login">
+  <div id="sign-in-widget"></div>
+
+  <script th:inline="javascript">
+      /*<![CDATA[*/
+
+      var config = {};
+
+      config.baseUrl = /*[[${oktaBaseUrl}]]*/ 'https://{yourOktaDomain}';
```

Comment:
Can we fulfill these from the environment variables we have from the backend side?

Reply:
The controller populates these values and sends it to the view. the controller uses the env variables/properties to originally fill in these values before passing it to the view.